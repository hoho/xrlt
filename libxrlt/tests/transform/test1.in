id:0, type:0, last:0, error: 0, data:
id:0, type:0, last:0, error: 0, data:
id:0, type:0, last:0, error: 0, data:
id:0, type:0, last:0, error: 0, data:
id:3, type:2, last:0, error: 0, data:{"aaa
id:3, type:2, last:1, error: 0, data:bbb": 123, "ccc": true}
id:0, type:0, last:0, error: 0, data:
id:2, type:2, last:0, error: 0, data:<hello>
id:2, type:2, last:0, error: 0, data:world
id:2, type:2, last:1, error: 0, data:</hello>
id:0, type:0, last:0, error: 0, data:
id:1, type:2, last:0, error: 0, data:there are
id:1, type:2, last:0, error: 0, data: many clouds
id:1, type:2, last:1, error: 0, data: in the sky
id:0, type:0, last:0, error: 0, data:
