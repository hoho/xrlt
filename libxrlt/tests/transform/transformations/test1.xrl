<?xml version="1.0"?>
<xrl:requestsheet xmlns:xrl="http://xrlt.net/Transform">

    <xrl:transformation name="xsl" type="xslt" src="test1.xsl">
        <xrl:param name="papapa1" />
        <xrl:param name="papapa2" />
        <xrl:param name="papapa3" select="'pep&amp;epe'"/>
        <xrl:param name="papapa4" />
    </xrl:transformation>
    <xrl:transformation name="xsl2" type="xslt-stringify" src="test1.xsl" />

    <xrl:response>
        <xrl:transform name="xsl">
            <xrl:with-param name="papapa1" select="9876" />
            <xrl:with-param name="papapa2">'alala'</xrl:with-param>
            <test>yoyoyo</test>
            <test>yaya</test>
        </xrl:transform>
        <xrl:text>|||</xrl:text>
        <xrl:transform name="xsl2">
            <quack>opop</quack>
            <xrl:include>
                <xrl:href>/heck</xrl:href>
                <xrl:type>text</xrl:type>
                <xrl:success>
                    <heck>
                        <xrl:value-of select="/" />
                    </heck>
                </xrl:success>
            </xrl:include>
            <quack>apap</quack>
        </xrl:transform>
    </xrl:response>

</xrl:requestsheet>
