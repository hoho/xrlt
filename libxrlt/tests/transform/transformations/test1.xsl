<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">

    <xsl:output method="html" />

    <xsl:template match="/">
        <xsl:text>hihi|</xsl:text>
        <xsl:for-each select="*">
            <xsl:value-of select="concat(local-name(), ': ', text())" />
            <xsl:text>|</xsl:text>
        </xsl:for-each>
        <xsl:text>haha</xsl:text>
    </xsl:template>

</xsl:stylesheet>
