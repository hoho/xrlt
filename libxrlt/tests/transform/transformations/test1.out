XRLT_STATUS_SUBREQUEST
sr id: 1
sr method: GET
sr type: TEXT
sr url: /heck
sr query: (null)
sr body: (null)
XRLT_STATUS_CHUNK
chunk: hihi|test: yoyoyo|test: yaya|haha
chunk: |||
XRLT_STATUS_WAITING
XRLT_STATUS_CHUNK
chunk: <yoyo>hihi|quack: opop|heck: quack-quack|quack: apap|haha</yoyo>

XRLT_STATUS_DONE
